testsuite/tests/readwrite2.rb

```diff
@@ -146,73 +150,79 @@ def main
         }
       }
 
+      # Change fstab name for test environment
+      Nfs.etc_fstab_name = FSTAB_NAME
+
+      # Using run_test from test_helper.rb to set up a temporary fstab before
+      # and dump it after each test
+      #
       # fstab contains nfs mounts & services are running
       DUMP("\nRead  - nfs is in use & running\n")
-      TEST(->() { Nfs.Read }, [@READ, @WRITE, @EXECUTE], nil)
+      run_test(->() { Nfs.Read }, [@READ, @WRITE, @EXECUTE], nil)
       # fstab unchanged
       DUMP("\nWrite - nfs is in use - start services\n")
-      TEST(->() { Nfs.Write }, [@READ, @WRITE, @EXECUTE], nil)
+      run_test(->() { Nfs.Write }, [@READ, @WRITE, @EXECUTE], nil)
 
       # fstab contains nfs mounts & services are stopped:-(
       DUMP("\nRead  - nfs is in use & stopped\n")
-      TEST(->() { Nfs.Read }, [@READ3, @WRITE, @EXECUTE], nil)
+      run_test(->() { Nfs.Read }, [@READ3, @WRITE, @EXECUTE], nil)
       # fstab unchanged - so, start services
       DUMP("\nWrite - nfs is in use - so, start services\n")
-      TEST(->() { Nfs.Write }, [@READ3, @WRITE, @EXECUTE], nil)
+      run_test(->() { Nfs.Write }, [@READ3, @WRITE, @EXECUTE], nil)
 
       # fstab contains no nfs mounts, services are running
       DUMP("\nRead  - nfs not used & running\n")
-      TEST(->() { Nfs.Read }, [@EMPTY, @WRITE, @EXECUTE], nil)
+      run_test(->() { Nfs.Read }, [@EMPTY, @WRITE, @EXECUTE], nil)
       # fstab unchanged - so, STOP services
       DUMP("\nWrite - nfs not used - so, stopping services\n")
-      TEST(->() { Nfs.Write }, [@EMPTY, @WRITE, @EXECUTE], nil)
+      run_test(->() { Nfs.Write }, [@EMPTY, @WRITE, @EXECUTE], nil)
 
       # fstab contains no nfs mount, serives are stopped
       DUMP("\nRead  - nfs not used & services are stopped\n")
-      TEST(->() { Nfs.Read }, [@EMPTY3, @WRITE, @EXECUTE], nil)
+      run_test(->() { Nfs.Read }, [@EMPTY3, @WRITE, @EXECUTE], nil)
       # fstab unchanged - so, leave services stopped
       DUMP("\nWrite - nfs not used; leave services stopped\n")
-      TEST(->() { Nfs.Write }, [@EMPTY3, @WRITE, @EXECUTE], nil)
+      run_test(->() { Nfs.Write }, [@EMPTY3, @WRITE, @EXECUTE], nil)
 
       # // nfs and portmap are running
```

Comment:
Um, maybe we should rather delete this dead code instead of updating...

Reply:
AFAICS it's not completely empty, but it tests that /etc/fstab remains unchanged and that the services are really stopped. From its expected stdout (file `tmp.out.readwrite2`):

    Dump        
    Dump        Write - nfs not used; leave services stopped
    Dump        
    Dump        tmp.fstab before:
    File        /dev/hda6  /  ext3  defaults  1  2
    Execute     .target.bash "/bin/cp $ORIG $BACKUP" $["BACKUP":"/etc/fstab.YaST2.save", "ORIG":"/etc/fstab"] 0
    Write       .sysconfig.nfs.NFS4_SUPPORT "no" true
    Write       .sysconfig.nfs.NFS_SECURITY_GSS "no" true
    Return      true
    Dump        tmp.fstab after:
    File        /dev/hda6  /  ext3  defaults  1  2

Similar to the test before:


    Dump        
    Dump        Read  - nfs not used & services are stopped
    Dump        
    Dump        tmp.fstab before:
    File        /dev/hda6  /  ext3  defaults  1  2
    Read        .sysconfig.nfs.NFS4_SUPPORT "no"
    Read        .sysconfig.nfs.NFS_SECURITY_GSS "no"
    Read        .etc.idmapd_conf.value.General.Domain "localhost"
    Return      true
    Dump        tmp.fstab after:
    File        /dev/hda6  /  ext3  defaults  1  2

Reply:
Ooops, sorry for confusion, the comment belongs to the commented block below, not to the displayed change which is OK.